### ISSUE TITLE
[Feature] Support per-step loss logging during training

### ISSUE THREAD

**Issue by GiGiKoneti** [mllam/neural-lam#674]
Currently, during training over multiple autoregressive steps, the overall training loss is aggregated into a single scalar and logged as `train_loss`. There is no way to monitor the degradation of loss at individual unroll steps during training (unlike validation and testing, which support per-step loss logging using `val_loss_unroll{step}`).

This enhancement proposes adding a `--train_steps_to_log` CLI option (similar to `--val_steps_to_log`) to allow users to log the training loss for specific unroll steps separately (e.g. `train_loss_unroll1`, `train_loss_unroll5`, `train_loss_unroll10`).
